test/k8sT/Policies.go

```diff
@@ -827,78 +736,157 @@ EOF`, k, v)
-						From: []networkingv1.NetworkPolicyPeer{{
-							NamespaceSelector: &metav1.LabelSelector{
-								MatchLabels: map[string]string{
-									"ns-name": namespace,
-								},
-							},
-						}},
-					}},
-				},
+		It("Tests the same Policy in the different namespaces", func() {
+
+			By("Applying Policy in %q namespace", namespace)
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, policy, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "L3 Policy cannot be applied in %q namespace", namespace)
+
+			By("Applying Policy in default namespace")
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, l3Policy, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "L3 Policy cannot be applied in %q namespace", helpers.DefaultNamespace)
+
+			By("Testing %q namespace", namespace)
+
+			res := kubectl.ExecPodCmd(
+				namespace, appPodsNS[helpers.App2],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIPNS)))
+			res.ExpectSuccess("%q cannot curl service", appPods[helpers.App2])
+
+			res = kubectl.ExecPodCmd(
+				namespace, appPodsNS[helpers.App3],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIPNS)))
+			res.ExpectFail("%q can curl to service", appPods[helpers.App3])
+
+			By("Testing default namespace")
+
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("%q cannot curl clusterIP %q", appPods[helpers.App2], clusterIP)
+
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App3],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectFail("%q can curl to %q", appPods[helpers.App3], clusterIP)
+		})
+
+		It("Network Policy namespace selector", func() {
+			By("Testing connectivity across Namespaces without policy")
+			for _, pod := range []string{helpers.App2, helpers.App3} {
+				res := kubectl.ExecPodCmd(
+					helpers.DefaultNamespace, appPods[pod],
+					helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIPNS)))
+				res.ExpectSuccess("%q cannot curl service", appPods[pod])
+
+				res = kubectl.ExecPodCmd(
+					namespace, appPodsNS[pod],
+					helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIPNS)))
+				res.ExpectSuccess("%q cannot curl service", appPodsNS[pod])
+			}
+
+			By("Applying Policy in %q namespace", namespace)
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, netpolNsSelector, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "Policy cannot be applied")
+
+			for _, pod := range []string{helpers.App2, helpers.App3} {
+				res := kubectl.ExecPodCmd(
+					helpers.DefaultNamespace, appPods[pod],
+					helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIPNS)))
+				res.ExpectFail("%q can curl to service and it should not. NS policy should be applied", appPods[pod])
```

Comment:
NS policy should be applied --> policy restricts communication across namespaces (or something more specific).